### PR TITLE
Console mode for multiple game automation

### PR DESCRIPTION
**Pull request by alkorolyov**
Hi,
As I said previously one of my plans was to run multiple games. I want to start working on it. Ideally I would like to get rid of libgdx dependence on game start. First step would be to decouple all loading/saving routines from game process itself.
I remeber adding ruleset to GameSetupInfo during work on scenario, but now understand it was not very wise decision :)

### CODE REVIEW

**Comment by alkorolyov** [comment]
@yairm210 
Also I found around ~50 mentions of UncivGame in the core.logic and models packages. Do you think we need to get rid of them in order to be able to run in fully console mode? How would you implement the console mode?

File: core/src/com/unciv/UncivGame.kt [hunk 1]
```diff
@@ -23,7 +23,8 @@ class UncivGame(
         private val crashReportSender: CrashReportSender? = null,
         val exitEvent: (()->Unit)? = null,
         val cancelDiscordEvent: (()->Unit)? = null,
-        val fontImplementation: NativeFontImplementation? = null
+        val fontImplementation: NativeFontImplementation? = null,
+        val consoleMode: Boolean = false
```
Comment (yairm210) [inline core/src/com/unciv/UncivGame.kt:27]: There are too many parameters here.
Please turn this into a class, like UncivGameParameters or something.

File: core/src/com/unciv/logic/GameSaver.kt [hunk 2]
```diff
@@ -57,7 +57,11 @@ object GameSaver {
     }
 
     fun getGeneralSettingsFile(): FileHandle {
-        return Gdx.files.local(settingsFileName)
+        try {
+            return Gdx.files.local(settingsFileName)
+        } catch (ex: NullPointerException) {
+            return FileHandle(settingsFileName)
```
Comment (yairm210) [inline core/src/com/unciv/logic/GameSaver.kt:63]: Why are you having local files replaces with filehandles everywhere? I must be missing something

File: core/src/com/unciv/UncivGame.kt (hunk 1 above)
Comment (alkorolyov) [reply core/src/com/unciv/UncivGame.kt:27]: Ok, undesrtood.

File: core/src/com/unciv/logic/GameSaver.kt (hunk 2 above)
Comment (alkorolyov) [reply core/src/com/unciv/logic/GameSaver.kt:63]: I was thinking to run in console mode without starting libgdx module Gdx.
And for all file IO we are using Gdx.files which is initialized after Gdx.app start.
For full console I was just importing FileHandle which doesn't need Gdx.app running it's just a wrapper for Java IO.
In such way I don't need to run LwjglApplication(game, config) and can make console mode more lightweight and quick.

Probably would be better to change to smth like to make it clear:
```
        if (UncivGame.consoleMode) {
            return FileHandle(settingsFileName)
        } else {
            return Gdx.files.local(settingsFileName)
```

**Comment by alkorolyov** [comment]
I have changed UncivGame initialization, tested at Desktop and Console versions. But please test Android and IOS versions because I am afraid to break something.

**Comment by yairm210** [comment]
Why are all the translation file changes marked as changes here? 🤔

**Comment by alkorolyov** [comment]
> Why are all the translation file changes marked as changes here? 🤔

Hmm, I don't know. I was just merging periodically with master branch ...

**Comment by alkorolyov** [comment]
That's good now, merged with the latest master.

**Comment by yairm210** [comment]
One change that needs to be made, now that I'm looking at the code inside - please change all try/catch of file loading to if(UncivGame.Current.consoleMode) ... else ...
That way it's a lot more obvious what we're checking for, and we don't have useless try/catches cluttering up our regular runtime

**Pull request merged**